### PR TITLE
accumulator: add error handling for writing to buffer

### PR DESCRIPTION
**Pull request by florianl**
Errors were ignored so far when trying to write to the buffer.

### CODE REVIEW

File: accumulator/batch.go [hunk 1]
```diff
@@ -195,7 +199,11 @@ func (b *Batch) AddAgentData(apmData APMData) error {
 	// first line being metadata.
 	data, after, _ := bytes.Cut(raw, newLineSep)
 	if b.metadataBytes == 0 {
-		b.metadataBytes, _ = b.buf.Write(data)
+		var bufErr error
+		b.metadataBytes, bufErr = b.buf.Write(data)
+		if bufErr != nil {
+			fmt.Errorf("failed to write data to buffer: %v", err)
```
Comment (florianl) [inline accumulator/batch.go:205]:
```suggestion
			return fmt.Errorf("failed to write data to buffer: %v", err)
```

File: accumulator/batch.go [hunk 2]
```diff
@@ -195,7 +199,11 @@ func (b *Batch) AddAgentData(apmData APMData) error {
 	// first line being metadata.
 	data, after, _ := bytes.Cut(raw, newLineSep)
 	if b.metadataBytes == 0 {
-		b.metadataBytes, _ = b.buf.Write(data)
+		var bufErr error
+		b.metadataBytes, bufErr = b.buf.Write(data)
+		if bufErr != nil {
+			return fmt.Errorf("failed to write data to buffer: %v", err)
```
Comment (rockdaboot) [inline accumulator/batch.go:205]:
```suggestion
			return fmt.Errorf("failed to write data to buffer: %v", bufErr)
```

File: accumulator/batch.go [hunk 3]
```diff
@@ -151,7 +151,11 @@ func (b *Batch) OnAgentInit(reqID, contentEncoding string, raw []byte) error {
 	b.mu.Lock()
 	defer b.mu.Unlock()
 	if b.metadataBytes == 0 && len(metadata) > 0 {
-		b.metadataBytes, _ = b.buf.Write(metadata)
+		var bufErr error
```
Comment (rockdaboot) [inline accumulator/batch.go:154]: Why `bufErr` and not re-using `err`?

File: accumulator/batch.go [hunk 4]
```diff
@@ -151,7 +151,11 @@ func (b *Batch) OnAgentInit(reqID, contentEncoding string, raw []byte) error {
...
+		b.metadataBytes, bufErr = b.buf.Write(metadata)
+		if bufErr != nil {
```
Comment (florianl) [inline accumulator/batch.go:156]:
```suggestion
		b.metadataBytes, err = b.buf.Write(metadata)
		if err != nil {
```

File: accumulator/batch.go [hunk 5]
```diff
@@ -195,7 +199,11 @@ func (b *Batch) AddAgentData(apmData APMData) error {
 	// first line being metadata.
 	data, after, _ := bytes.Cut(raw, newLineSep)
 	if b.metadataBytes == 0 {
-		b.metadataBytes, _ = b.buf.Write(data)
+		var bufErr error
+		b.metadataBytes, bufErr = b.buf.Write(data)
+		if bufErr != nil {
```
Comment (florianl) [inline accumulator/batch.go:204]:
```suggestion
		b.metadataBytes, err = b.buf.Write(data)
		if err != nil {
```

File: accumulator/batch.go (hunk 2 above)
Comment (florianl) [reply accumulator/batch.go:203]: applied the suggestion

File: accumulator/batch.go [hunk 6]
```diff
@@ -151,7 +151,10 @@ func (b *Batch) OnAgentInit(reqID, contentEncoding string, raw []byte) error {
 	b.mu.Lock()
 	defer b.mu.Unlock()
 	if b.metadataBytes == 0 && len(metadata) > 0 {
-		b.metadataBytes, _ = b.buf.Write(metadata)
+		b.metadataBytes, err = b.buf.Write(metadata)
+		if err != nil {
+			return fmt.Errorf("failed to write metadata to buffer: %v", bufErr)
```
Comment (florianl) [inline accumulator/batch.go:156]:
```suggestion
			return fmt.Errorf("failed to write metadata to buffer: %v", err)
```

**Review by rockdaboot — APPROVED** [review]
LGTM

(nit: merge `err != nil` into previous line)

**Pull request merged**